### DISCUSSION TITLE
Rounded edges in 2d and 3d (again)

### DISCUSSION THREAD

**Discussion by macbryla** [bevyengine/bevy#9671] [Q&A] [answered]
I am new to Bevy and am looking at Bevy as a solution for data visualisation and potentially visual simulation projects in the future. For data visualisations to work, they need to be expressive, intuitive, and beautiful. It's the beautiful part that I have a few questions about.

Let me give you an example of an existing data visualisation I created a while back. It is written using Egui and runs in wasm. I attached a screenshot, and you can see the live version here (https://macbryla.com/data/aus-energy-transition/viz/index.html), which auto-updates with the latest data every 5 minutes. 
This all works fine, and I will explain later why moving from Egiu to Bevy interests me.

I tried to recreate the above example in Bevy, but as far as I know, I cannot create 2D rectangles with rounded edges. It seems like a minor issue, but in my example, it makes the visualisation much more visually pleasing. I also assume that the same limitation applies to 3D cubes, which can only have sharp edges.

Is this feature on the way anytime soon? Or is there a solution to reproduce the above in Bevy? I have seen some discussion but am unsure where that is today.

I want to move from Egui to Bevy for this work to better animate some of the transitions, as this is a live data viz. I also think that a 3D option using cubes and rotating them as new data becomes available every 5 minutes. This visualisation lives on a massive screen, and a visual indicator when data changes would be great. So, the view would still look 2D, but the transitions would be 3D.

Egui is brilliant at what it does, but I think it would be much cleaner to code a richer visualisation with smooth transitions in Bevy.

I would be very interested in your thoughts, and maybe I am missing some Bevy features that would allow me to do this today. 
Thanks all.

<img width="1430" alt="Screenshot 2023-06-25 at 11 13 52 am" src="https://github.com/bevyengine/bevy/assets/251217/9cc96b03-3a15-48b5-be40-d02385b620ef">

**Comment by rparrett** [comment, marked as answer]
There's no built-in helper for creating these shapes, but all the machinery exists to build [meshes](https://docs.rs/bevy/latest/bevy/prelude/struct.Mesh.html#manual-creation) with round corners or any shape you want.

There are plenty of options if you don't want to do that though!

Check out out the [shapes section](https://bevyengine.org/assets/#shapes) of Bevy Assets as well as [bevy_prototype_lyon](https://github.com/Nilirad/bevy_prototype_lyon) which isn't in that category for some reason and recently added support for rounded corners. WASM support in these crates may vary.

There's also a [PR in flight](https://github.com/bevyengine/bevy/pull/8973) that adds a corner radius option to UI nodes.

**Reply by macbryla** [reply]
Brilliant, thank you. 
Yes, I should have looked at the plugins.
